invenio_app_ils/ui/src/invenio_app_ils/pages/frontsite/PatronProfile/PatronPastLoans/PatronPastLoans.js

```diff
@@ -0,0 +1,141 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {
+  Loader,
+  Error,
+  Pagination,
+} from '../../../../common/components';
+import {toShortDate} from '../../../../common/api/date';
+import {invenioConfig} from '../../../../common/config';
+import {Container, Grid, Header, Item, Label} from "semantic-ui-react";
+import {getCover} from "../../config";
+import {Link} from "react-router-dom";
+import {FrontSiteRoutes} from "../../../../routes/urls";
+import {DocumentAuthors} from "../../components/Document";
+import {
+  ILSHeaderPlaceholder,
+  ILSParagraphPlaceholder
+} from "../../../../common/components/ILSPlaceholder";
+import {ILSItemPlaceholder} from "../../../../common/components/ILSPlaceholder/ILSPlaceholder";
+
+
+class PastLoanListEntry extends Component {
+
+  render() {
+    const {loan} = this.props;
+    return (
+      <Item key={loan.metadata.pid}>
+        <Item.Image size='mini'
+                    src={getCover(loan.metadata.document_pid)}
+                    as={Link}
+                    disabled
+                    to={FrontSiteRoutes.documentDetailsFor(
+                      loan.metadata.document_pid)}
+        />
+
+        <Item.Content>
+          <Item.Header as={Link}
+                       to={FrontSiteRoutes.documentDetailsFor(
+                         loan.metadata.document_pid)}>
+            {loan.metadata.document.title}</Item.Header>
+          <Grid columns={2}>
+            <Grid.Column mobile={16} tablet={8} computer={8}>
+              <Item.Meta>
+                <DocumentAuthors metadata={loan.metadata.document}/>
+                Loaned on {toShortDate(loan.metadata.start_date)}
+              </Item.Meta>
+              <Item.Description>
+                {}
+                You have extended this loan {loan.metadata.extension_count}
+                {' '}of {invenioConfig.loans.max_extensions_count} times
+              </Item.Description>
+            </Grid.Column>
+            <Grid.Column textAlign={'right'}
+                         mobile={16} tablet={8} computer={8}
+            >
+              <Item.Description>
+                Literature returned on {' '}
+                <Label>
+                  {toShortDate(loan.metadata.end_date)}
+                </Label>
+              </Item.Description>
+            </Grid.Column>
+          </Grid>
+        </Item.Content>
+
+      </Item>
+    )
+  }
+}
+
+
+export default class PatronPastLoans extends Component {
+  constructor(props) {
+    super(props);
+    this.fetchPatronPastLoans = this.props.fetchPatronPastLoans;
+    this.patronPid = this.props.patronPid;
+    this.state = {activePage: 1};
+  }
+
+  componentDidMount() {
+    this.fetchPatronPastLoans(this.patronPid);
```

Comment:
I am sorry, big change warning here :)
I think that in all these components, you don't need shared state between components, so no need for Redux. WDYT?

Reply:
I'm sharing the state with 3 of the five of them, for the stats count. I would leave them like this as it might be used for displaying the summary